src/common/page_table.h

```diff
@@ -48,6 +45,43 @@ struct SpecialRegion {
  * mimics the way a real CPU page table works.
  */
 struct PageTable {
+    static constexpr int ATTRIBUTE_BITS = 2;
+
+    class PageInfo {
+    public:
+        [[nodiscard]] u8* Pointer() const noexcept {
+            return ExtractPointer(raw);
+        }
+
+        [[nodiscard]] PageType Type() const noexcept {
+            return ExtractType(raw);
+        }
+
+        [[nodiscard]] std::pair<u8*, PageType> PointerType() const noexcept {
+            const uintptr_t non_atomic_raw = raw;
```

Comment:
```suggestion
            const uintptr_t non_atomic_raw = raw.load(std::memory_order_relaxed);
```

If all that matters is atomicity

Ditto for usages in `Pointer()` and `Type()` as well